dev-lang/vala/vala38-0.38.8.recipe

```diff
@@ -0,0 +1,131 @@
+SUMMARY="An object-oriented programming language that uses the GObject system"
+DESCRIPTION="Vala is a new programming language that aims to bring modern \
+programming language features to GNOME developers without imposing any \
+additional runtime requirements and without using a different ABI \
+compared to applications and libraries written in C."
+HOMEPAGE="https://wiki.gnome.org/Projects/Vala"
+COPYRIGHT="2006-2018 Jürg Billeter
+	2006-2018 Raffaele Sandrini"
+LICENSE="GNU LGPL v2.1"
+REVISION="1"
+SOURCE_URI="https://download.gnome.org/sources/vala/${portVersion%.*}/vala-$portVersion.tar.xz"
+CHECKSUM_SHA256="2fa746b51cd66e43577d1da06a80b708c2875cadaafee77e9700ea35cf23882c"
+SOURCE_DIR="vala-$portVersion"
+PATCHES="vala38-$portVersion.patchset"
+
+libVersion="0.0.0"
+libVersionCompat="$libVersion compat >= ${libVersion%%.*}"
+
+ARCHITECTURES="!x86_gcc2 ?x86 ?x86_64"
+SECONDARY_ARCHITECTURES="?x86"
+
+PROVIDES="
+	vala$secondaryArchSuffix = $portVersion compat >= 0
+	cmd:vala$secondaryArchSuffix
+	cmd:valac$secondaryArchSuffix
+	cmd:valadoc$secondaryArchSuffix
+	cmd:vapicheck$secondaryArchSuffix
+	cmd:vapigen$secondaryArchSuffix
+	cmd:vala_gen_introspect$secondaryArchSuffix
+	cmd:vala_${portVersion%.*}$secondaryArchSuffix
+	cmd:vala_gen_introspect_${portVersion%.*}$secondaryArchSuffix
+	cmd:valac_${portVersion%.*}$secondaryArchSuffix
+	cmd:valadoc_${portVersion%.*}$secondaryArchSuffix
+	cmd:vapicheck_${portVersion%.*}$secondaryArchSuffix
+	cmd:vapigen_${portVersion%.*}$secondaryArchSuffix
+	lib:libvala_${portVersion%.*}$secondaryArchSuffix = $libVersionCompat
+	lib:libvaladoc_${portVersion%.*}$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libdbus_1$secondaryArchSuffix
+	lib:libffi$secondaryArchSuffix
+	lib:libglib_2.0$secondaryArchSuffix
+	lib:libgvc$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	lib:libintl$secondaryArchSuffix
+	"
+
```

Comment:
Among the 70 recipes in dev-python, 22 have one or more `REPLACES`, and most of these 22 recipes do have a blank line before their `REPLACES`, but these are special because they follow a `for .. in .. do .. done` to handle our multiple python versions.

If we ignore the dev-python category, all other recipes that have a `REPLACES` do not have an empty line before it.

So I'd say that we don't need a blank line between `REQUIRES` and `REPLACES`, just like we don't have a blank line between `PROVIDES` and `REQUIRES`.

Reply:
I think the previous version 0.36.12 is still being developed, would it be wise to also include vala in the REPLACES section for this version? (don't know if both can exist at  the same time ...

Reply:
We don't need to have `REPLACES` for this purpose, because pkgman knows that:
* it needs to uninstall `0.36.12` if you try to install `0.38.8`, and
* it needs to uninstall `0.38.8` if you try to install `0.36.12`.

Reply:
`REPLACES` is only used if the basename of a package changes. Here, both recipes provide the same package name: vala.

Oops, I just noticed that you named your recipe `vala38-0.38.8.recipe`. With this name you would need to have a different `PROVIDES`:
```diff
-	vala$secondaryArchSuffix = $portVersionCompat
+	vala38$secondaryArchSuffix = $portVersionCompat
```
and use a `CONFLICTS`.
`REPLACES` are stronger than `CONFLICTS` because once you ask pkgman to install something, it automatically uninstalls any package which is in the `REPLACES`.
`REPLACES` should only be used when you wish to retire an old package name or get rid of old subpackages.

For vala, you can probably rename `vala38-0.38.8.recipe` to `vala-0.38.8.recipe` and let the users choose the version they want.
BTW, if `0.38.8` is not yet well tested then it is better to mark it as untested. That's OK because `0.36.12` will be available from the binary repos.